### PR TITLE
Fix: filter statuses by project

### PR DESCRIPTION
**Pull request by lewtakm**
This PR adds projectKey variable to function that fetch Statuses in order to filter them correctly per project.
Also small change was included that removes FC from plugin. 
Related issue: https://github.com/RoadieHQ/backstage-plugin-jira/issues/8

### CODE REVIEW

**Comment by adamdmharvey** [comment]
This is awesome, thanks for the PR, @lewtakm !

I gave this a shot locally, and unfortunately it doesn't look quite right.

image
image

My fault, I told you the JSON response was identical between global and project, and looks like it's not.  So the root `/status` shows you EVERY status in the ENTIRE system.  project/KEY/statuses shows you the status as applicable per issue type.

So for example, it is possible to say give an issue type of "EPIC" as "OPEN, ASSESSING, SCHEDULED, IN PROGESS, IN REVIEW, DONE" while "Bug" issueType could instead be "OPEN, TRIAGING, IN PROGRESS, TESTING, DONE".  So the status Json inside of each issue type section is the same - a list of resulting status-  but in fact it COULD be different per issue type.  (not often is, but CAN be)

I think for the purposes of the plugin that's ok, in that we could collect all available statuses for all issue types, present them, and if I filter for status TRIAGING then fine, I get only bugs back and that's fine.

See vendor doco here:

https://docs.atlassian.com/software/jira/docs/api/REST/8.14.0/#api/2/project-getAllStatuses

And sample is roughly:

```
 {
        "self": "http://localhost:8090/jira/rest/api/2.0/issueType/3",
        "id": "3",
        "name": "Task",
        "subtask": false,
        "statuses": [
            {
                "self": "http://localhost:8090/jira/rest/api/2.0/status/10000",
                "description": "The issue is currently being worked on.",
                "iconUrl": "http://localhost:8090/jira/images/icons/progress.gif",
                "name": "In Progress",
                "id": "10000"
            },
            {
                "self": "http://localhost:8090/jira/rest/api/2.0/status/5",
                "description": "The issue is closed.",
                "iconUrl": "http://localhost:8090/jira/images/icons/closed.gif",
                "name": "Closed",
                "id": "5"
            }
        ]
    }
```

So I think a minor tweak to how that Json is mapped is needed?

**Review by martina-if — DISMISSED** [review]

**Comment by lewtakm** [comment]
I just updated the PR in order to adapt to different response type.

File: src/hooks/useActivityStream.ts [hunk 1]
```diff
@@ -36,32 +43,43 @@ export const useActivityStream = (size: number) => {
...
-          icon: {
-            url: icon.href,
-            title: icon.title,
-          },
-          summary: decodeHtml(getPropertyValue(entry, 'summary') || ''),
-          content: decodeHtml(getPropertyValue(entry, 'content') || ''),
-        }
-      });
+      const parsedData = JSON.parse(
+        convert.xml2json(response, { compact: true, spaces: 2 }),
```
Comment (iain-b) [inline src/hooks/useActivityStream.ts:47]: Can we not request json via the http headers on the request? e.g. 
```
Content-Type: application/json
Accept: application/json
```

**Comment by iain-b** [comment]
Would be nice to see a quick loom given there are a lot of changes? Nice  work though! thanks!

**Comment by lewtakm** [comment]
> Would be nice to see a quick loom given there are a lot of changes? Nice work though! thanks!

@iain-b there are lots of changes due to running prettier. In fact the onle part has been actually really changed is getStatuses() function, so widget looks the same as earlier. 
chrome_qEPYJ6LQGN

File: src/hooks/useActivityStream.ts (hunk 1 above)
Comment (lewtakm) [reply src/hooks/useActivityStream.ts:47]: It's an RSS feed in XML taken directly from Jira,  I think it won't be possible to get is a JSON format directly.

**Review by iain-b — APPROVED** [review]

**Review by iain-b — APPROVED** [review]

**Pull request merged**